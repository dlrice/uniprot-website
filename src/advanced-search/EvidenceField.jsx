import React, { Fragment } from 'react';

// type Props = {
//   updateEvidence: Function,
//   selectedEvidence?: string,
//   url: string,
//   data: Array<Group>,
// };

const EvidenceField = ({ value = '', handleChange, data = [] }) => {
  if (!data) {
    return null;
  }

  return (
    <Fragment>
      <label htmlFor="evidence_select">
        Evidence
        <select id="evidence_select" value={value} onChange={e => handleChange(e.target.value)}>
          {data.map(group => (
            <optgroup label={group.groupName} key={group.groupName}>
              {group.items.map(item => (
                <option value={item.code} key={item.code}>
                  {item.name}
                </option>
              ))}
            </optgroup>
          ))}
        </select>
      </label>
    </Fragment>
  );
};

export default EvidenceField;
