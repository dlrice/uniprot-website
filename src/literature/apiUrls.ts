import { joinUrl, devPrefix } from '../utils/apiUrls';

const literatureApiUrls = {
  literature: joinUrl(devPrefix, '/uniprot/api/literature'),
};

export const getPublicationURL = (id: string) =>
  joinUrl(literatureApiUrls.literature, id);

export const getPublicationsURL = (ids: string[]) =>
  `${literatureApiUrls.literature}/searchs?query=(${ids
    .map((id) => `id:${id}`)
    .join(' OR ')})`;

export default literatureApiUrls;
