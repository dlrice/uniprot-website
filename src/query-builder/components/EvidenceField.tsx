import React, { useEffect, useState } from 'react';
import { QueryBit, SearchTermType } from '../types/searchTypes';
import initializer from '../utils/fieldInitializer';

const EvidenceField: React.FC<{
  field: SearchTermType;
  handleChange: (queryBit: QueryBit) => void;
  initialValue?: QueryBit;
}> = ({ field, handleChange, initialValue }) => {
  const [value, setValue] = useState(() => initializer(field, initialValue));

  useEffect(() => {
    if (value.length > 0) {
      handleChange({ [field.id]: `(${field.term}:${value?.trim()})` });
    }
  }, [field, value, handleChange]);

  return (
    <label htmlFor="evidence_select">
      Evidence
      <select
        id="evidence_select"
        data-testid="evidence-select"
        value={value}
        onChange={(e) => setValue(e.target.value)}
      >
        {field.evidenceGroups?.map((group) => (
          <optgroup label={group.groupName} key={group.groupName}>
            {group.items.map((item: { code: string; name: string }) => (
              <option value={item.code} key={item.code}>
                {item.name}
              </option>
            ))}
          </optgroup>
        ))}
      </select>
    </label>
  );
};

export default EvidenceField;
