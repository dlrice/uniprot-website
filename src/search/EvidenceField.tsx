import React, { Fragment } from 'react';

type EvidenceFieldProps = {
  value: string | undefined;
  handleChange: (value: string) => void;
  data: any[];
};

const EvidenceField: React.FC<EvidenceFieldProps> = ({ value = '', handleChange, data = [] }) => {
  if (!data) {
    return null;
  }

  return (
    <Fragment>
      <label htmlFor="evidence_select">
        Evidence
        <select id="evidence_select" value={value} onChange={e => handleChange(e.target.value)}>
          {data.map(group => (
            <optgroup label={group.groupName} key={group.groupName}>
              {group.items.map((item: { code: string; name: string }) => (
                <option value={item.code} key={item.code}>
                  {item.name}
                </option>
              ))}
            </optgroup>
          ))}
        </select>
      </label>
    </Fragment>
  );
};

export default EvidenceField;
