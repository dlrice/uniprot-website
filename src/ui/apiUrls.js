const prefix = '//wwwdev.ebi.ac.uk/uniprot/api';

export default {
  // uniprotkb advanced search terms
  advanced_search_terms: `${prefix}/configure/uniprotkb/search_terms`,
  // Annotation evidence used by advanced search
  annotation_evidences: `${prefix}/configure/uniprotkb/annotation_evidences`,
  // Go evidences used by advanced go search
  // "itemType": "goterm",
  go_evidences: `${prefix}/configure/uniprotkb/go_evidences`,
  // Database cross references used by advanced search
  database_xefs: `${prefix}/configure/uniprotkb/databases`,
  // Database cross reference fields in result column configure
  // "itemType": "database",
  database_fields: `${prefix}/configure/uniprotkb/databasefields`,
  // All result fields except database cross reference fields
  results_fields: `${prefix}/configure/uniprotkb/resultfields`,
};
