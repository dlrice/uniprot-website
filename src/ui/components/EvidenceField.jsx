import React from 'react';

// type Props = {
//   updateEvidence: Function,
//   selectedEvidence?: string,
//   url: string,
//   data: Array<Group>,
// };

const EvidenceField = ({ field, value='', handleChange, data=[] }) => {
  if (!data) {
    return null;
  }

  return (
    <div className="advanced-search__inputs">
      <label htmlFor="evidence_select">
        Evidence
        <select
          id="evidence_select"
          value={value}
          onChange={e => handleChange(e.target.value)}
        >
          {data.map(group => (
            <optgroup label={group.groupName} key={group.groupName}>
              {group.items.map(item => (
                <option value={item.code} key={item.code}>
                  {item.name}
                </option>
              ))}
            </optgroup>
          ))}
        </select>
      </label>
    </div>
  );
}

export default EvidenceField;
