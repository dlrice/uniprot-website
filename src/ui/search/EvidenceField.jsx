// @flow
import React, { Component } from 'react';
import withData from '../hoc/withData';

type Props = {
  updateEvidence: Function,
  selectedEvidence?: string,
  url: string,
  data: Array<Group>,
};

class EvidenceField extends Component<Props> {
  static defaultProps = {
    selectedEvidence: '',
  };

  render() {
    const { data, selectedEvidence, updateEvidence } = this.props;
    if (!data) {
      return null;
    }
    return (
      <div className="advanced-search__inputs">
        <label htmlFor="evidence_select">
          Evidence
          <select id="evidence_select" value={selectedEvidence} onChange={e => updateEvidence(e)}>
            {data.map(group => (
              <optgroup label={group.groupName} key={group.groupName}>
                {group.items.map(item => (
                  <option value={item.code} key={item.code}>
                    {item.name}
                  </option>
                ))}
              </optgroup>
            ))}
          </select>
        </label>
      </div>
    );
  }
}

export default withData(props => props.url)(EvidenceField);
