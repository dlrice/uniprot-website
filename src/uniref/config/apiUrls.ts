import queryString from 'query-string';
import joinUrl from 'url-join';

export const devPrefix = 'http://wp-np2-be:8091';
export const prodPrefix = 'https://www.ebi.ac.uk';

const apiUrls = {
  search: joinUrl(devPrefix, 'uniprot/api/uniref/search'),
};

export default apiUrls;

export const getAPIQueryUrl = (query: string) => {
  return `${apiUrls.search}?${queryString.stringify({
    query: `${query}`,
  })}`;
};
