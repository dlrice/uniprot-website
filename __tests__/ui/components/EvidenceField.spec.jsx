import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import apiUrls from '../../../src/ui/apiUrls';
import EvidenceField from '../../../src/ui/components/EvidenceField';

configure({ adapter: new Adapter() });

describe('EvidenceField component', () => {
  test('should render', () => {
    const component = renderer
      .create(<EvidenceField url={apiUrls.annotation_evidences} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should fire change function on selection', () => {
    const updateEvidence = jest.fn(() => {});
    const wrapped = mount(
      <EvidenceField url={apiUrls.annotation_evidences} updateEvidence={updateEvidence} />,
    );
    wrapped
      .find('#evidence_select')
      .first()
      .simulate('change');
    expect(updateEvidence).toHaveBeenCalled();
  });

  test('should set selectedEvidence in the props', () => {
    const selectedEvidence = 'c1';
    const wrapped = shallow(
      <EvidenceField url={apiUrls.annotation_evidences} selectedEvidence={selectedEvidence} />,
    );
    expect(wrapped.instance().props.selectedEvidence).toBe(selectedEvidence);
  });
});
