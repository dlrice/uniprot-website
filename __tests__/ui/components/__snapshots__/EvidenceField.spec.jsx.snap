// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvidenceField component should render 1`] = `
"<div className=\\"advanced-search__inputs\\">
  <label htmlFor=\\"evidence_select\\">
    Evidence
    <select id=\\"evidence_select\\" value=\\"\\" onChange={[Function: onChange]}>
      <optgroup label=\\"foo\\">
        <option value={0}>
          bar
        </option>
        <option value={1}>
          baz
        </option>
      </optgroup>
    </select>
  </label>
</div>"
`;
